src/main/java/pl/connectis/Spotify_API_CLI/authorization/AuthorizationCodeAuthorization.java

```diff
@@ -0,0 +1,127 @@
+import org.springframework.context.ApplicationContext;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.MediaType;
+import org.springframework.stereotype.Component;
+import org.springframework.util.LinkedMultiValueMap;
+import org.springframework.util.MultiValueMap;
+import org.springframework.web.client.RestTemplate;
+import org.springframework.web.util.UriComponentsBuilder;
+
+import java.io.IOException;
+import java.util.Base64;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.locks.Condition;
+import java.util.concurrent.locks.ReentrantLock;
+
+@Slf4j
+@Component("authorization_code")
+public class AuthorizationCodeAuthorization {
+
+    private final AuthorizationCode authorizationCode;
+    private final ApplicationContext context;
+    private final RestTemplate restTemplate;
+    private final HttpHeaders httpHeaders;
+    private final ReentrantLock lock;
+    private final Condition condition;
+    @Value("${authorizationMethod}")
+    private String authorizationMethod;
+    @Value("${clientId}")
+    private String clientID;
+    @Value("${clientSecret}")
+    private String clientSecret;
+    @Value("${userAuthorizationUri}")
+    private String userAuthorizationUri;
+    @Value("${accessTokenUri}")
+    private String accessTokenUri;
+    @Value("{$spring.main.web-application-type}")
+    private String webApplicationType;
+
+
+    public AuthorizationCodeAuthorization(AuthorizationCode authorizationCode, ApplicationContext context, RestTemplate restTemplate, HttpHeaders httpHeaders, ReentrantLock lock, Condition condition) {
+        this.authorizationCode = authorizationCode;
+        this.context = context;
+        this.restTemplate = restTemplate;
+        this.httpHeaders = httpHeaders;
+        this.lock = lock;
+        this.condition = condition;
+    }
+
+    public void authorize() throws IOException {
+
+        if (!webApplicationType.equals("servlet")) {
+            throw new RuntimeException("Wrong spring web application type configuration for authorization_code method of authorization. Please change the method or set spring.main.web-application-type to servlet.");
+        }
+
+        HttpEntity<MultiValueMap<String, String>> httpEntity;
+
+        UriComponentsBuilder builder = UriComponentsBuilder.fromHttpUrl(userAuthorizationUri)
+                .queryParam("client_id", clientID)
+                .queryParam("response_type", "code")
+                .queryParam("redirect_uri", "http://localhost:8080/collect");
+//				.queryParam("state", "")
+//				.queryParam("scope", "")
+//				.queryParam("show_dialog", "")
+
+        log.info(builder.build().toUri().toString());
+
+        try {
+            java.awt.Desktop.getDesktop().browse(builder.build().toUri());
+        } catch (IOException ex) {
+            log.error("Couldn't open default browser.");
+            ex.printStackTrace();
+        }
+
+        lock.lock();
```

Comment:
po co Ci ten lock? nie widze opcji żeby autoryzacja odpaliła się kilkukrotnie żeby więc nie widze potrzeby stosowania sekcji krytycznej

Reply:
Lock jest, aby poczekać na reakcję użytkownika w przypadku autoryzacji `authorization code flow` i żeby mi nie uciekł wtedy główny wątek wykonujący kod w `SpotifyAPICLR`.

Reply:
blokowanie całego wątku to bardzo zły pomysł. Locki potrzebne są do synchronizacji pamięci wątków a nie orkiestrowania nimi. Możesz np zastosować spring eventy i kiedy event o poprawnej autoryzacji się wygeneruje to wtedy w listenerze kontunuujesz działanie aplikacji.